File: 1-easy/01827-minimum-operations-to-make-the-array-increasing/solution.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        count = 0
        for i in range(1, len(nums)):
            if nums[i - 1] >= nums[i]:
                amount_to_add = nums[i - 1] - nums[i] + 1
                count += amount_to_add
                nums[i] += amount_to_add

        return count

File: 1-easy/01827-minimum-operations-to-make-the-array-increasing/solution.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
        int count = 0;
        for (int i{1}; i < nums.size(); ++i) {
            if (nums[i - 1] >= nums[i]) {
                int amount_to_add = nums[i - 1] - nums[i] + 1;
                count += amount_to_add;
                nums[i] += amount_to_add;
            }
        }

        return count;
    }
};
